Index: sys/arch/arm/fdt/a9ptmr_fdt.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/fdt/a9ptmr_fdt.c,v
retrieving revision 1.7
diff -u -p -r1.7 a9ptmr_fdt.c
--- sys/arch/arm/fdt/a9ptmr_fdt.c	5 Nov 2022 17:30:06 -0000	1.7
+++ sys/arch/arm/fdt/a9ptmr_fdt.c	25 Mar 2025 02:34:23 -0000
@@ -81,10 +81,15 @@ a9ptmr_fdt_attach(device_t parent, devic
 	struct fdt_attach_args * const faa = aux;
 	const int phandle = faa->faa_phandle;
 	bus_space_handle_t bsh;
-	uint32_t mpidr;
+	uint32_t mpidr, rate;
+	char intrstr[128];
 	bool is_hardclock;
 
 	sc->sc_dev = self;
+	prop_dictionary_t dict = device_properties(self);
+	if (prop_dictionary_get_uint32(dict, "frequency", &rate))
+		goto skip;
+
 	sc->sc_clk = fdtbus_clock_get_index(phandle, 0);
 	if (sc->sc_clk == NULL) {
 		aprint_error(": couldn't get clock\n");
@@ -95,11 +100,9 @@ a9ptmr_fdt_attach(device_t parent, devic
 		return;
 	}
 
-	uint32_t rate = clk_get_rate(sc->sc_clk);
-	prop_dictionary_t dict = device_properties(self);
+	rate = clk_get_rate(sc->sc_clk);
 	prop_dictionary_set_uint32(dict, "frequency", rate);
-
-	char intrstr[128];
+skip:
 	if (!fdtbus_intr_str(phandle, 0, intrstr, sizeof(intrstr))) {
 		aprint_error(": failed to decode interrupt\n");
 		return;
@@ -162,6 +165,11 @@ a9ptmr_fdt_speed_changed(device_t dev)
 	prop_dictionary_t dict = device_properties(dev);
 	uint32_t rate;
 
+	if (sc->sc_clk == NULL) {
+		aprint_error(": couldn't get clock\n");
+		return;
+	}
+
 	rate = clk_get_rate(sc->sc_clk);
 	prop_dictionary_set_uint32(dict, "frequency", rate);
 
