Index: sys/arch/arm/ti/files.ti
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/ti/files.ti,v
retrieving revision 1.22
diff -u -p -r1.22 files.ti
--- sys/arch/arm/ti/files.ti	20 Jul 2022 10:01:11 -0000	1.22
+++ sys/arch/arm/ti/files.ti	25 Mar 2025 02:39:10 -0000
@@ -3,13 +3,19 @@
 
 file	arch/arm/ti/ti_cpufreq.c	soc_ti
 file	arch/arm/ti/am3_platform.c	soc_am33xx
-file	arch/arm/ti/omap3_platform.c	soc_omap3
+file	arch/arm/ti/omap3_platform.c	soc_omap3 | soc_omap4
+file	arch/arm/ti/omap_smc.S		soc_omap4
 
 # Interrupt controller
 device	omapintc: pic, pic_splfuncs
 attach  omapintc at fdt
 file	arch/arm/ti/ti_omapintc.c	omapintc
 
+# WakeupGen
+device	omapwugen
+attach	omapwugen at fdt with omapwugen
+file	arch/arm/ti/ti_omapwugen.c	omapwugen
+
 # sysc interconnect
 device	tisysc { } : fdt, ti_prcm
 attach	tisysc at fdt with ti_sysc
@@ -34,6 +40,11 @@ device	omap3prm { } : fdt
 attach	omap3prm at fdt with omap3_prm
 file	arch/arm/ti/omap3_prm.c		omap3_prm
 
+# PRCM (OMAP4)
+device	omap4prcm { } : fdt, ti_prcm
+attach	omap4prcm at fdt with omap4_prcm
+file	arch/arm/ti/omap4_prcm.c	omap4_prcm
+
 # Clocks
 device	timuxclk
 attach	timuxclk at fdt with ti_mux_clock
@@ -47,6 +58,10 @@ device	tidpllclk
 attach	tidpllclk at fdt with ti_dpll_clock
 file	arch/arm/ti/ti_dpll_clock.c	ti_dpll_clock
 
+device	tigateclk
+attach	tigateclk at fdt with ti_gate_clock
+file	arch/arm/ti/ti_gate_clock.c	ti_gate_clock
+
 # UART
 attach	com at fdt with ti_com: ti_prcm
 file	arch/arm/ti/ti_com.c		ti_com needs-flag
@@ -142,3 +157,4 @@ file	arch/arm/ti/ti_wdt.c		ti_wdt
 defflag	opt_soc.h			SOC_TI
 defflag	opt_soc.h			SOC_AM33XX: SOC_TI
 defflag	opt_soc.h			SOC_OMAP3: SOC_TI
+defflag	opt_soc.h			SOC_OMAP4: SOC_TI
