Index: sys/dev/fdt/fixedfactorclock.c
===================================================================
RCS file: /cvsroot/src/sys/dev/fdt/fixedfactorclock.c,v
retrieving revision 1.4
diff -u -p -r1.4 fixedfactorclock.c
--- sys/dev/fdt/fixedfactorclock.c	27 Jan 2021 03:10:21 -0000	1.4
+++ sys/dev/fdt/fixedfactorclock.c	7 Apr 2025 01:56:12 -0000
@@ -120,8 +120,8 @@ fixedfactorclock_attach(device_t parent,
 	clk_attach(&sc->sc_clk.base);
 
 	aprint_naive("\n");
-	aprint_normal(": x%u /%u fixed-factor clock\n",
-	    sc->sc_clk.mult, sc->sc_clk.div);
+	aprint_normal(": x%u /%u fixed-factor clock (%s)\n",
+	    sc->sc_clk.mult, sc->sc_clk.div, name);
 
 	fdtbus_register_clock_controller(self, phandle,
 	    &fixedfactorclock_fdt_funcs);
